CONTRIBUTING.md

```diff
@@ -1,112 +1,124 @@
+privileged user. As a result, the included systemd service runs as `root`.
 
-*MQTT broker*: Needed to publish messages. [Mosquitto](https://mosquitto.org/) is easy to set up and an excellent broker that can be run offline and locally. The online [solution](https://test.mosquitto.org/) can be used but is not recommended.
+## `yggd`
 
+Make sure `yggd` is running and listening on the system bus. If using a
+`yggdrasil` package, starting the `yggdrasil.service` unit should be sufficient.
+Be sure to configure `/etc/yggdrasil/config.toml` to connect to an MQTT broker,
+should you wish to send messages over the network.
 
-*HTTP server*: An optional HTTP server used to request payloads from localhost. This does not need to be more complicated than using the Python's `SimpleHTTPServer` module.
+## `mqttcli`
 
-# Install rhc-worker-playbook develop setup
+It is recommended to install a simple MQTT publish/subscribe utility. Included
+in Fedora is `mqttcli`, a pair of simple client utilities for such purpose.
 
-`rhc-worker-playbook` needs some devel packages in order to compile the C extensions, which may take some time to run as it dowloads and install everything needed to the `rhc-worker-playbook`.
+## HTTP server
 
-```console
-$ sudo dnf install c-ares-devel openssl-devel python3-devel gcc gcc-c++
-$ python -m venv .venv
-$ source .venv/bin/activate
-$ make build
-```
+An HTTP server is used to request payloads from localhost. This does not need to
+be more complicated than using the Python's `SimpleHTTPServer` module.
+
+# Compilation
 
-Additionally it can be used through `rhc` installing the `rhc-worker-playbook` as a worker.
+It is possible, though awkward, to run `rhc-worker-playbook` directly from the
+project repository. If you're working on a Fedora-derived distribution, use the
+included `srpm` meson target to create a SRPM. This SRPM can then be built using
+`mock` or `koji` or any other RPM build system.
 
 ```console
-$ sudo make CONFIG_DIR=$(pkg-config rhc --variable workerconfdir) installed-lib-dir
-$ sudo make build
-$ sudo make install CONFIG_DIR=$(pkg-config rhc --variable workerconfdir)
+meson setup -Dbuild_srpm=True builddir
+meson compile srpm -C builddir
+mock --rebuild ./builddir/dist/srpm/*.src.rpm
 ```
 
 # Test environment Quickstart
 
-This environment recreates the `rhc-worker-playbook`, in which an `ansible playbook` is sent through the MQTT broker, `yggdrasil` processes it and send it to the worker to execute it.
+This environment recreates the `rhc-worker-playbook`, in which an `ansible
+playbook` is sent through the MQTT broker, `yggdrasil` processes it and send it
+to the worker to execute it.
 
 
-**TERMINAL 1**
+## TERMINAL 1
 
 Run an *HTTP server* that will serve the payload. Then run `mosquitto` service.
 
 ```console
-$ nohup python3 -m http.server 8000 > /dev/null 2>&1 &
-$ mosquitto
+$ nohup python3 -m http.server --directory ./testdata 8000 > /dev/null 2>&1 &
```

Comment:
I would not recommend to start this server with `nohup` and redirecting everything to `/dev/null` I could be useful to see result of this simple http server.